src/wallet/wallet.js

```diff
@@ -262,10 +293,36 @@ export class Wallet {
   encryptRecoveryData (kdfParams, recoveryKeypair) {
     let recoveryWallet = new Wallet(
       this.email,
-      this._keypair,
+      this._signingKeypair,
       recoveryKeypair.accountId()
     )
 
     return recoveryWallet.encrypt(kdfParams, recoveryKeypair.secret())
   }
+
+  /**
+   * Set signing Keypair
+   *
+   * @param {Keypair} keypair signing Keypair.
+   */
+  useSigningKeypair (keypair) {
+    if (this._isValidKeypair(keypair) && isString(keypair)) {
```

Comment:
same here - inside `else` block you don't check for keypair being a valid. If I pass an object - everything will be okay, and error will be thrown only after trying to sign smth with it